ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -369,9 +394,18 @@ class JITModBuilder {
     }
   }
 
-  void handlePHINode(Instruction *I, BasicBlock *BB) {
+  void handlePHINode(Instruction *I, BasicBlock *BB, size_t CurBBIdx, size_t CurInstrIdx) {
     Value *V = ((PHINode *)&*I)->getIncomingValueForBlock(BB);
-    VMap[&*I] = getMappedValue(V);
+    // From the point of view of the compiled trace it's enough to just update the `Vmap` with the
+    // incoming value `V`. However, this means there's no corresponding instruction in the JITMod
+    // and thus there's nothing mapping back to it during a guard failure. Creating a
+    // (non-sensical) select instruction here makes sure there's a mapping from a JITMod value back
```

Comment:
I don't think `nonsensical` is the right word. Maybe `degenerate`. You might spin the comment in the direction of "an assignment would work, but LLVM doesn't have one, so we emulate one with a degenerate select statement"?

Reply:
I think perhaps the main concept we can want to get across is "this select instruction doesn't do anything bad". What the right term for that is I'm not immediately sure!